pom.xml

```diff
@@ -121,6 +121,8 @@
         <module>presto-product-tests-launcher</module>
         <module>presto-jdbc</module>
         <module>presto-cli</module>
+        <module>presto-cli-provisio</module>
```

Comment:
Does the module name look okay? Presto server was broken down as:

`presto-server` - provisio assembly descriptor, since `presto-cli` was already taken I chose `presto-cli-provisio`.
`presto-server-main` - actual presto server module equivalent to `presto-cli`
`presto-server-rpm` - rpm module for presto server equivalent to `presto-cli-rpm`

Reply:
Can we have presto-cli-main (the current presto-cli renamed), presto-cli that would produce presto-cli-executable and provisio package and then presto-cli-rpm? That way we will be coherent with presto-server. WDYT?

Reply:
It will be coherent but since it was an invasive/breaking change (people expect presto-cli to contain code today) I did not feel confident enough to take that decision.
If you feel it is ok, I can re-arrange accordingly.

Reply:
Does it mean artifact names for presto-cli would change? eg links to maven central?

Reply:
@findepi Yes. The artifactId will change.

Reply:
> The artifactId will change.

you mean for CLI code, or for final executable?

Reply:
The existing executable JAR that gets created will change and the CLI code. The RPM artifact will remain the same.

Reply:
> The existing executable JAR that gets created will change

That's a bummer. Can we avoid that?

Reply:
I don't think so. Only the suffix is configurable. The artifact name will always contain the `artifactId` AFAIK. Maybe @jvanzyl might have some ideas.

Reply:
Another option is to rename `presto-cli-provisio` --> `presto-cli-assembly` since that will at-least make the intent clear and be familiar to people from other projects.
Unfortunately renaming `presto-cli` would definitely mean breaking any scripts that download the artifact from Maven central.

Reply:
i don't want this inline conversation to 'disappear', so -> https://github.com/prestosql/presto/pull/4025#issuecomment-650175037

Reply:
Maybe if building cli RPM takes few seconds we could simply inline all modules into one? Is that even possible?

Reply:
Sadly that doesn't work due to reasons outlined in [here](https://github.com/prestosql/presto/pull/4025/files#diff-4ba198ba64f21a442f8a95c010e07ffbR2-R6). I'd really like for that comment being proven false though. :cry:

Reply:
For me it's fine to have:
* `presto-cli-assembly`
* `presto-cli-rpm`
* `presto-cli`
but we should add a comment why this extra modules are needed in their `pom.xml`s